x-pack/legacy/plugins/graph/public/render_app.ts

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiConfirmModal } from '@elastic/eui';
+
+// inner angular imports
+// these are necessary to bootstrap the local angular.
+// They can stay even after NP cutover
+import angular from 'angular';
+import { i18nDirective, i18nFilter, I18nProvider } from '@kbn/i18n/angular';
+import 'ui/angular-bootstrap';
+import 'ace';
+import 'ui/kbn_top_nav';
+import { configureAppAngularModule } from 'ui/legacy_compat';
+// @ts-ignore
+import { createTopNavDirective, createTopNavHelper } from 'ui/kbn_top_nav/kbn_top_nav';
+// @ts-ignore
+import { confirmModalFactory } from 'ui/modals/confirm_modal';
+
+// type imports
+import { DataStart } from 'src/legacy/core_plugins/data/public';
+import {
+  AppMountContext,
+  ChromeStart,
+  SavedObjectsClientContract,
+  ToastsStart,
+  UiSettingsClientContract,
+} from 'kibana/public';
+// @ts-ignore
+import { initGraphApp } from './app';
+import { Plugin as DataPlugin } from '../../../../../src/plugins/data/public';
+
+/**
+ * These are dependencies of the Graph app besides the base dependencies
+ * provided by the application service. Some of those still rely on non-shimmed
+ * plugins in LP-world, but if they are migrated only the import path in the plugin
+ * itself changes
+ */
+export interface GraphDependencies extends LegacyAngularInjectedDependencies {
+  element: HTMLElement;
+  appBasePath: string;
+  capabilities: Record<string, boolean | Record<string, boolean>>;
+  coreStart: AppMountContext['core'];
+  chrome: ChromeStart;
+  config: UiSettingsClientContract;
+  toastNotifications: ToastsStart;
+  indexPatterns: DataStart['indexPatterns']['indexPatterns'];
+  npData: ReturnType<DataPlugin['start']>;
+  savedObjectsClient: SavedObjectsClientContract;
+  xpackInfo: { get(path: string): unknown };
+  addBasePath: (url: string) => string;
+  getBasePath: () => string;
+  Storage: any;
+  canEditDrillDownUrls: boolean;
+  graphSavePolicy: string;
+}
+
+/**
+ * Dependencies of the Graph app which rely on the global angular instance.
+ * These dependencies have to be migrated to their NP counterparts.
+ */
+export interface LegacyAngularInjectedDependencies {
+  /**
+   * angular $http service
+   */
+  $http: any;
+  /**
+   * Instance of SavedObjectRegistryProvider
+   */
+  savedObjectRegistry: any;
+  kbnBaseUrl: any;
+  /**
+   * Instance of SavedWorkspacesProvider
+   */
+  savedGraphWorkspaces: any;
+}
+
+export const renderApp = ({ appBasePath, element, ...deps }: GraphDependencies) => {
+  const graphAngularModule = createLocalAngularModule(deps.coreStart);
+  configureAppAngularModule(graphAngularModule);
```

Comment:
Really happy this turned out useful...spent a lot of time breaking this out only to wonder if we would use it.

That said, we need to spend some time removing some of the legacy dependencies that this module uses. Doesn't look like there's too much:

```
import { fatalError } from 'ui/notify';
import { capabilities } from 'ui/capabilities';
import { modifyUrl } from 'ui/url';
import { UrlOverflowService } from '../error_url_overflow';
import { toastNotifications } from '../notify';
import { isSystemApiRequest } from '../system_api';
```

We should also make this function take CoreStart as an argument rather than importing it via `ui/new_platform`.

Reply:
I have to check but it might be sufficient to just pull in the necessary parts - it looks like Graph only really needs the badges handling and even that could probably be moved to the controller. It's probably not so simple for the Kibana app apps though, so still worth thinking about this.

Reply:
I worked on this in the dashboard shim PR - are you fine with leaving this as it is for now and we'll fix it there?